### DISCUSSION TITLE
Available Tests

### DISCUSSION THREAD

**Discussion by PowerfulTitan** [siliconcompiler/siliconcompiler#2563] [Help (Q&A)] [answered]
Hi!
I am trying to see what are the unit tests, the system level tests, the tool-checking tests, etc. I was wondering if these were all the tests available: https://github.com/siliconcompiler/siliconcompiler/blob/main/tests/README.md and the breakdown regarding the type of tests they are?

**Comment by gadfort** [comment]
@PowerfulTitan I'm not sure I follow your question. The tests are "somewhat" organized according to that document. Anything that is marked the `eda` requires the tools to be available, otherwise they should all be python tests. Is there something specific you are looking for?

**Reply by Subashkatel** [reply]
@gadfort I have a question related to this: What do the core tests comprise of? For example, if I want to verify if just a single RTL module functionality is working or not based on some pre-written tests/test-bench, is this possible with the current test or not?

**Reply by gadfort** [reply, marked as answer]
@Subashkatel anything in the tests directory is just for unit testing siliconcompiler itself, it would not be a good place to try to test some RTL. I would suggest you look in the examples folder for how to verify your RTL: https://github.com/siliconcompiler/siliconcompiler/blob/main/examples/gcd/gcd_sta.py
and 
https://github.com/siliconcompiler/siliconcompiler/blob/main/examples/heartbeat/heartbeat_sim.py

**Reply by Subashkatel** [reply]
@gadfort Thank you for that! Do you know of any other resources where I can do these tests?

**Reply by gadfort** [reply]
@Subashkatel without a clearer understanding of the task you are trying to accomplish, I'm not sure I can provide guidance.

**Reply by mi-and-core** [reply]
@PowerfulTitan  @Subashkatel @gadfort  It's a bit off-topic, but if you're interested in testing your systemverilog, I recommend you take a look at Universal Verification Methodology (UVM, IEEE 1800.2 standard) and pyuvm. As the name suggests, pyuvm uses Python (Cocotb is used to interact with simulators).

pyuvm ver. 3.0.0 (with UVM RAL support) was released a few hours ago: https://github.com/pyuvm/pyuvm/releases/tag/3.0.0

If you're new to UVM, see https://github.com/pyuvm/pyuvm/blob/master/README.md for a quick ALU-test.
